drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/collectioneditor/CollectionPresenter.java

```diff
@@ -70,6 +70,11 @@
      */
     protected Map<String, Map<String, Map<String, String>>> expandablePropertiesMap = new HashMap<>();
 
+    /**
+     * It defines if <b>Create collection</b> guided editor is in editing status or not.
+     */
+    protected boolean createCollectionIsEditing = false;
```

Comment:
What about define an enum to keep the status of the whole panel? Something like `GUIDED`, `EXPRESSION` and `GUIDED_ADDING_ELEMENT`. I think it is easier to understand